Dockerfile

```diff
@@ -1,27 +1,21 @@
 FROM ubuntu:xenial
 
-MAINTAINER OPS <ops@nelen-schuurmans.nl>
-
-# Change the date to force rebuilding the whole image
-ENV REFRESHED_AT 20160531
-
 # system dependencies
-RUN apt-get update && apt-get install -y \
-    python-software-properties \
-    wget \
+RUN apt-get update && apt-get install -y --no-install-recommends \
     build-essential \
-    git \
-    libevent-dev \
-    libfreetype6-dev \
-    libpng12-dev \
     python3-dev \
     python3-pip \
-    python3-psycopg2 \
+    python3-venv \
+    python3-wheel \
     gettext \
     postgresql-client \
 && apt-get clean -y && rm -rf /var/lib/apt/lists/*
 
-RUN pip3 install -U zc.buildout setuptools
-
-VOLUME /code
 WORKDIR /code
+
+RUN python3 -m venv .
+COPY . .
```

Comment:
- Het gebruik van een virtualenv in een Docker is niet gebruikelijk.
- Je doet nu een COPY in de Dockerfile én een mount in de docker-compose. Het is of-of :)
- Als je een COPY doet, dan kunnen er wat directories ge-ignored worden middels een `.dockerignore`. Ik zou in ieder geval de `var` dir ignoren. Ik geloof dat je alleen `requirements.txt` nodig hebt om de docker te bouwen.
- Waarom doe je collectstatic? is deze docker ook voor productie? doen we dat op dit moment ook?

Reply:
Virtualenv komt wel voor, maar da's dan vooral om de dependencies heel netjes gecontroleerd te krijgen. Maar dat is in dit geval niet nodig, want het is gewoon een library "waarvan alleen de tests maar hoeven te runnen". => die ga ik veranderen.

Reply:
De copy + mount is wel nodig (misschien in een andere vorm, ik zat er weer even mee te experimenteren). Er is een setup.py waarin de dependencies staan. requirements.txt heeft zo'n `-e .` regel, dus er moet op dat moment al wel wat in de docker zitten, namenlijk de setup.py en alles wat daarin genoemd staat.

Als ik de virtualenv weghaal kan het weer ietsje simpeler.

De mount is vervolgens om wel lokaal te kunnen devven.

Reply:
var en Collectstatic: de var zit alleen in de docker (anders moet die ook gemount worden).

Omdat de var in de docker zit draai ik collectstatic om 'em te vullen. Betekent helaas wel een "build" als je iets veranderd, maar dat gebeurd toch de komende paar jaar niet meer.

Deze docker is trouwens alleen om te ontwikkelen (en dus vooral om makkelijk een goede python versie en een postgres te krijgen).

Reply:
Oke! Nou de dockerfile is weer net een ander smaakje dan ik in andere repos heb gezien. Er zijn altijd weer lastige vraagstukken op te lossen als je het op een andere manier gaat doen. Maar in principe is het OK.

Twee dingen die ik me nog afvraag:
-  kan het nu gebeuren dat python als "root" (want in docker) bestanden gaat aanmaken in de mount `./lizard_auth_server:/code/lizard_auth_server` (bijv. `.pyc` files of `__pycache__`)? Misschien goed om de benadering die in veel repos wordt gebruikt te doen: (https://github.com/nens/cookiecutter-djangosite-template/blob/d317ed15a2784a9211898edce2973df5db86892d/%7B%7Bcookiecutter.project_slug%7D%7D/Dockerfile#L44)
- ik zie geen `-e .` in the requirements.txt staan. Is ook niet nodig wat mij betreft. Maar ik denk dat t met alleen een `COPY requirements.txt` misschien ook wel werkt

Reply:
De "-e" staat er toch echt: https://github.com/lizardsystem/lizard-auth-server/pull/89/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552R1

En hij is 100% nodig, want het is een library en dan moet je toch echt je dependencies specificeren. Bij een site kom je er nog net mee weg, maar hier niet.

Reply:
Dat uid/gid trucje zal ik toevoegen.
(... heb ik zelf geen last van i.v.m. mac, daarom vergeet ik het steeds ...)

Reply:
Wat ik nu ook zie: ik heb automatisch die "pip install" toegevoegd, maar het wordt allemaal makkelijker als je dat van buiten af zelf aan moet roepen. Het hoort een beetje bij het ontwikkelproces, dus is het helemaal niet verkeerd om dat te moeten doen.

Dan ben je gelijk van de `COPY` af.